object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionVisitor.java

```diff
@@ -61,6 +92,73 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
     return super.visitMethodInvocation(node, p);
   }
 
+  @Override
+  public Void visitNewClass(NewClassTree node, Void p) {
+    if (!isAssignedToLocal(this.getCurrentPath())) {
+      TypeMirror type = TreeUtils.typeOf(node);
+      if (hasAlwaysCall(type)) {
+        checker.report(node, new DiagMessage(Diagnostic.Kind.ERROR, "missing.alwayscall", " "));
+      }
+    }
+    return super.visitNewClass(node, p);
+  }
+
+  private List<String> getCalledMethodAnnotation(MethodInvocationTree node) {
+    AnnotationMirror calledMethodAnno;
+    AnnotatedTypeMirror currentType = getTypeFactory().getAnnotatedType(node);
+
+    if (currentType == null || !currentType.isAnnotatedInHierarchy(atypeFactory.TOP)) {
+      calledMethodAnno = atypeFactory.TOP;
+    } else {
+      calledMethodAnno = currentType.getAnnotationInHierarchy(atypeFactory.TOP);
+    }
+
+    return getValueOfAnnotationWithStringArgument(calledMethodAnno);
+  }
+
+  private boolean isAssignedToLocal(final TreePath treePath) {
+    TreePath parentPath = treePath.getParentPath();
+
+    if (parentPath == null) {
+      return false;
+    }
+
+    Tree parent = parentPath.getLeaf();
+    switch (parent.getKind()) {
+      case PARENTHESIZED:
+        return isAssignedToLocal(parentPath);
+      case CONDITIONAL_EXPRESSION:
+        ConditionalExpressionTree cet = (ConditionalExpressionTree) parent;
+        if (cet.getCondition() == treePath.getLeaf()) {
+          // The assignment context for the condition is simply boolean.
+          // No point in going on.
+          return false;
+        }
+        // Otherwise use the context of the ConditionalExpressionTree.
+        return isAssignedToLocal(parentPath);
+      case ASSIGNMENT: // check if the left hand is a local variable
+        // TODO! not sure about this part!
+        final JCTree.JCExpression lhs = ((JCTree.JCAssign) parent).lhs;
+        if (!(lhs instanceof JCTree.JCFieldAccess)) {
+          return (((JCTree.JCIdent) lhs).sym.getKind() == LOCAL_VARIABLE);
```

Comment:
Rather than casting to `JCIdent` no matter what if this isn't a field access, I think we should be defensive and check whether the class, and throw a `BugInCF` exception if it's not `JCIdent`